fc/fc_app.py

```diff
@@ -565,6 +647,79 @@ def run(index,
         _LOG.info("Runner finished normally, triggering shutdown.")
     finally:
         runner.stop()
+    return 0
+
+
+def all_files_exist(filesnames: list):
+    """
+    Return True if all files in a list exist.
+
+    :param filesnames: A list of file paths.
+    :return:
+    """
+    isthere = [os.path.isfile(i) for i in filesnames]
+    return all(isthere)
+
+
+def filename2tif_names(filename: Path, bands: list, sep='_'):
+    """
+    Turn one file name into several file names, one per band.
+    This turns a .tif filename into a dictionary of filenames,
+    the band as the key, with the band inserted into the file names.
+        i.e ls8_fc.tif -> ls8_fc_BS.tif  (Last underscore is separator)
+
+    :param filename: a Path.
+    :param bands: a list of bands/measurements
+    :param sep: the separator between the base name and the band.
+    :return: filenames
+    """
+    base, ext = os.path.splitext(filename)
+    assert ext == '.tif'
+    filenames = {}
+    for band in bands:
+        build = Path(base + sep + band + ext)
+        filenames[band] = build.absolute().as_uri()
```

Comment:
If this was relative to the yaml instead, we could avoid manipulating the file before S3 upload

Reply:
Good idea Andrew. I'll look into doing this.